### ISSUE TITLE
Creating parent directories on remote

### ISSUE THREAD

**Issue by itsjareds** [ProtractorNinja/cu-handin-magic-make#1]
Hey Austin, I was playing around with this for one of my Operating Systems projects. It looks great! Took me embarrassingly long to figure out how to upload my public key to the handin site.

I had a small issue where `rsync` failed in the `remote` target because the parent directory `~/$(REMOTE_TEST_DIRECTORY)/` did not exist. Two possible solutions to this are to either:

```
ssh $(REMOTE_USER)@access.cs.clemson.edu mkdir -p ~/$(REMOTE_TEST_DIRECTORY)/$(ASSIGNMENT)
```

or to [have rsync call `mkdir -p`](http://www.schwertly.com/2013/07/forcing-rsync-to-create-a-remote-path-using-rsync-path/) before starting:

```
RSYNC_OPTS = --exclude=".git" --exclude="$(LOCAL_HANDIN)/" --delete-delay -r . \
             --rsync-path="mkdir -p ~/$(REMOTE_TEST_DIRECTORY)/$(ASSIGNMENT)/ && rsync"
@rsync $(RSYNC_OPTS) $(REMOTE_USER)@access.cs.clemson.edu:~/$(REMOTE_TEST_DIRECTORY)/$(ASSIGNMENT)/
```

I like the second one a bit more since it all happens during one connection.

Great work!

**Comment by ProtractorNinja** [comment]
Nice catch. I'll work on a fix.

**Comment by ProtractorNinja** [comment]
It's reproducible! I got there by setting `REMOTE_TEST_DIRECTORY` to a subfolder lacking initialized parents.

**Comment by ProtractorNinja** [comment]
Should be fixed, as of 3034411225842ff108dc56932bebb41097d5b98f.

**Comment by itsjareds** [comment]
Awesome, thanks!
